pkg/jobrunaggregator/jobrunaggregatorlib/disruption.go

```diff
@@ -0,0 +1,196 @@
+package jobrunaggregatorlib
+
+import (
+	"encoding/json"
+	"fmt"
+	"math"
+	"regexp"
+	"strings"
+	"time"
+
+	"k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/openshift/ci-tools/pkg/junit"
+)
+
+var (
+	upgradeBackendNameToTestSubstring = map[string]string{
+		"kube-api-new-connections":                          "Kubernetes APIs remain available for new connections",
+		"kube-api-reused-connections":                       "Kubernetes APIs remain available with reused connections",
+		"openshift-api-new-connections":                     "OpenShift APIs remain available for new connections",
+		"openshift-api-reused-connections":                  "OpenShift APIs remain available with reused connections",
+		"oauth-api-new-connections":                         "OAuth APIs remain available for new connections",
+		"oauth-api-reused-connections":                      "OAuth APIs remain available with reused connections",
+		"service-load-balancer-with-pdb-reused-connections": "Application behind service load balancer with PDB is not disrupted",
```

Comment:
There's something wrong with this one.

Reused and new I think are part of the same test -- https://github.com/openshift/origin/blob/master/test/e2e/upgrade/service/service.go#L42

https://prow.ci.openshift.org/view/gs/origin-ci-test/logs/periodic-ci-openshift-release-master-ci-4.10-upgrade-from-stable-4.9-e2e-gcp-ovn-upgrade/1454701479095963648 failed on new connections, but it was persisted to bigquery as "service-load-balancer-with-pdb-reused-connections".

I think this needs to be renamed to `service-load-balancer-with-pdb` and the bigquery rows updated? @deads2k WDYT?

Reply:
The disruption is calculated in two steps.  Reused connections are here https://github.com/openshift/origin/blob/master/test/e2e/upgrade/service/service.go#L198-L230

new connections are here https://github.com/openshift/origin/blob/master/test/e2e/upgrade/service/service.go#L246-L278

both produce uniquely identifiable monitor events, which are then added individually by the code writing the backenddisruption json file.